tests/escapedHashtags.js

```diff
@@ -58,33 +69,32 @@ define([
                     ["#form/text1", "`#form/text1`"],
                     ["/data/text1", "`#form/text1`"],
                     ["`#form/text1`", "`#form/text1`"],
+
+                    ["#form/text1 -1", "`#form/text1` - 1"],
+                    ["/data/text1 -1", "`#form/text1` - 1"],
+                    // ideally no change, but too hard right now (to much extra parsing going on)
```

Comment:
I think some of that extra parsing was due to attempting to be smart about how we stored the hashtag references in XML. For some reason we stored the hashtags unescaped when it was a valid xpath with known references, and escaped in the case of invalid xpaths. Do you remember the reason we decided to do that? Wouldn't storing escaped all the time be much simpler?